test/alternator/conftest.py

```diff
@@ -60,14 +60,7 @@ def pytest_collection_modifyitems(config, items):
         if "veryslow" in item.keywords:
             item.add_marker(skip_veryslow)
 
-# "dynamodb" fixture: set up client object for communicating with the DynamoDB
-# API. Currently this chooses either Amazon's DynamoDB in the default region
-# or a local Alternator installation on http://localhost:8080 - depending on the
-# existence of the "--aws" option. In the future we should provide options
-# for choosing other Amazon regions or local installations.
-# We use scope="session" so that all tests will reuse the same client object.
-@pytest.fixture(scope="session")
-def dynamodb(request):
+def _dynamodb(request):
```

Comment:
An idea: Instead of the ugly name "_dynamodb" which the user is not supposed to use (you created dynamodb_function_scoped instead), you could have a useful context manager, say "dynamodb_session", which the user can use directly and then you don't need dynamodb_function_scoped. In cql-pytest I did something like this - there is (in util.py) a cql_session() context manager, which tests can use directly (or through the new_session() wrapper) and the "cql" fixture also uses.

Reply:
looks like closing is not required in boto3, it closes connections when object is destructed, couldn't find precise docs about it but at least people on the internet say so...